egs2/TEMPLATE/tts1/tts.sh

```diff
@@ -1111,12 +1125,82 @@ else
     log "Skip the evaluation stages"
 fi
 
+if ! "${skip_scoring}"; then
+    if [ ${stage} -le 9 ] && [ ${stop_stage} -ge 9 ]; then
+        _gen_dir=${tts_exp}/${inference_tag}/${test_set}
+        _data=${data_feats}/${test_set}
+        if ! ${skip_wer}; then
+            ./scripts/utils/evaluate_asr.sh \
+                --whisper_tag ${whisper_tag} \
+                --whisper_dir ${whisper_dir} \
+                --cleaner ${cleaner} \
+                --hyp_cleaner ${hyp_cleaner} \
+                --inference_nj ${inference_nj} \
+                --nj ${nj} \
+                --gt_text ${_data}/text \
+                --gpu_inference ${gpu_inference} \
+                ${_gen_dir}/wav/wav_test.scp ${_gen_dir}/scoring/eval_wer
+
+            echo "Finished WER evaluation, results are in ${_gen_dir}/scoring/eval_wer"
+        fi
+
+        echo "Scoring TTS evaluation via VERSA, using default ${versa_config}. You can visit https://github.com/shinjiwlab/versa?tab=readme-ov-file#list-of-metrics for more supported metrics."
+        _opts=
+        _eval_dir=${_gen_dir}/scoring/versa_eval
+        mkdir -p ${_eval_dir}
+
+        _pred_file=${_gen_dir}/wav/wav_test.scp
+        _score_config=${versa_config}
+        _gt_file=${_data}/wav_test.scp
+
+        _nj=$(( ${inference_nj} < $(wc -l < ${_pred_file}) ? ${inference_nj} : $(wc -l < ${_pred_file}) ))
+
+        _split_files=""
+        for n in $(seq ${_nj}); do
+            _split_files+="${_eval_dir}/pred.${n} "
+        done
+        utils/split_scp.pl ${_pred_file} ${_split_files}
+
+        if [ -n "${_gt_file}" ]; then
+            _split_files=""
+            for n in $(seq ${_nj}); do
+                _split_files+="${_eval_dir}/gt.${n} "
+            done
+            utils/split_scp.pl ${_gt_file} ${_split_files}
+            _opts+="--gt ${_eval_dir}/gt.JOB"
+        fi
+
+        if ${gpu_inference}; then
+            _cmd="${cuda_cmd}"
+            _ngpu=1
+        else
+            _cmd="${decode_cmd}"
+            _ngpu=0
+        fi
+
+        ${_cmd} --gpu "${_ngpu}" JOB=1:"${_nj}" "${_eval_dir}"/versa_eval.JOB.log \
+        python -W ignore -m versa.bin.scorer \
```

Comment:
Thanks for the update! I think we can also include  wer evaluation here with the text information?